velodrome/transform/fetcher.go

```diff
@@ -26,103 +27,108 @@ import (
 )
 
 // fetchRecentIssues retrieves issues from DB, but only fetches issues modified since last call
-func fetchRecentIssues(db *gorm.DB, last time.Time, out chan sql.Issue) (time.Time, error) {
-	glog.Infof("Fetching issues updated after %s", last)
+func fetchRecentIssues(db *gorm.DB, last *time.Time, out chan sql.Issue) error {
+	glog.Infof("Fetching issues updated after %s", *last)
 
 	var issues []sql.Issue
-	query := db.Where("issue_updated_at >= ?", last).Preload("Labels").Find(&issues)
+	query := db.Where("issue_updated_at >= ?", last).Order("issue_updated_at").Preload("Labels").Find(&issues)
 	if query.Error != nil {
-		return last, query.Error
+		return query.Error
 	}
 
 	count := len(issues)
 	for _, issue := range issues {
 		out <- issue
-		last = issue.IssueUpdatedAt
+		*last = issue.IssueUpdatedAt
 	}
 	glog.Infof("Found and pushed %d updated/new issues", count)
 
-	return last, nil
+	return nil
 }
 
-// fetchRecentEvents retrieves events from DB, but only fetches events created since last call
-func fetchRecentEvents(db *gorm.DB, last int, out chan sql.IssueEvent) (int, error) {
-	glog.Infof("Fetching issue-events with id bigger than %d", last)
+// fetchRecentEventsAndComments retrieves events from DB, but only fetches events created since last call
+func fetchRecentEventsAndComments(db *gorm.DB, lastEvent *int, lastComment *int, out chan interface{}) error {
+	glog.Infof("Fetching issue-events with id bigger than %d", *lastEvent)
+	glog.Infof("Fetching comments with id bigger than %d", *lastComment)
 
-	rows, err := db.Model(sql.IssueEvent{}).Where("id > ?", last).Rows()
+	eventRows, err := db.Model(sql.IssueEvent{}).Where("id > ?", *lastEvent).Order("event_created_at asc").Rows()
 	if err != nil {
-		return last, err
-	}
-
-	count := 0
-	for rows.Next() {
-		var issueEvent sql.IssueEvent
-		db.ScanRows(rows, &issueEvent)
-		out <- issueEvent
-		last = issueEvent.ID
-		count++
+		return fmt.Errorf("Failed to query events from database: %s", err)
 	}
-	glog.Infof("Found and pushed %d new events", count)
-
-	return last, nil
-}
-
-// fetchRecentComments retrieves comments from DB, but only fetches comments created since last call
-func fetchRecentComments(db *gorm.DB, last int, out chan sql.Comment) (int, error) {
-	glog.Infof("Fetching comments with id bigger than %d", last)
 
-	rows, err := db.Model(sql.Comment{}).Where("id > ?", last).Rows()
+	commentRows, err := db.Model(sql.Comment{}).Where("id > ?", *lastComment).Order("comment_created_at asc").Rows()
 	if err != nil {
-		return last, err
+		return fmt.Errorf("Failed to query comments from database: %s", err)
 	}
 
 	count := 0
-	for rows.Next() {
-		var comment sql.Comment
-		db.ScanRows(rows, &comment)
-		out <- comment
-		last = comment.ID
+	var comment *sql.Comment = &sql.Comment{}
```

Comment:
Not sure what the preferred style is, but you can just say `comment := &sql.Comment{}`.